service/handlers.go

```diff
@@ -113,3 +126,47 @@ func (e *ExecuteStruct) GetClient(projectID string) (client.RancherClient, error
 	}
 	return *apiClient, nil
 }
+
+func saveWebhook(webhookRequestData map[string]interface{}) error {
+	config := config.GetConfig()
+	if _, ok := webhookRequestData["name"].(string); !ok {
+		return fmt.Errorf("Name not found after decode")
+	}
+	webhookName := webhookRequestData["name"].(string)
```

Comment:
You check if it is there and if it is a string and then you do it again on this line. You can do this just once thusly:
```
if webhookName, ok := webhookRequestData["name"].(string)
if !ok {
    return fmt.Errorf("Name not found after decode")
}
```

Based on my above comment, you'll be moving this out to the calling function. When you do, if the name is not present, make sure your return a 400 response code instead of 500. 400 is user error, 500 is server error